test/moves/protect.test.ts

```diff
@@ -26,92 +25,205 @@ describe("Moves - Protect", () => {
 
   beforeEach(() => {
     game = new GameManager(phaserGame);
+    game.override
+      .battleStyle("single")
+      .moveset([MoveId.PROTECT, MoveId.SPIKY_SHIELD, MoveId.ENDURE, MoveId.SPLASH])
+      .enemySpecies(SpeciesId.SNORLAX)
+      .enemyAbility(AbilityId.INSOMNIA)
+      .enemyMoveset(MoveId.LUMINA_CRASH)
+      .startingLevel(100)
+      .enemyLevel(100);
+  });
 
-    game.override.battleStyle("single");
+  it("should protect the user from attacks and their secondary effects", async () => {
+    await game.classicMode.startBattle([SpeciesId.CHARIZARD]);
 
-    game.override.moveset([MoveId.PROTECT]);
-    game.override.enemySpecies(SpeciesId.SNORLAX);
+    const charizard = game.scene.getPlayerPokemon()!;
 
-    game.override.enemyAbility(AbilityId.INSOMNIA);
-    game.override.enemyMoveset([MoveId.TACKLE]);
+    game.move.select(MoveId.PROTECT);
+    await game.phaseInterceptor.to("BerryPhase", false);
 
-    game.override.startingLevel(100);
-    game.override.enemyLevel(100);
+    expect(charizard.hp).toBe(charizard.getMaxHp());
+    expect(charizard.getStatStage(Stat.SPDEF)).toBe(0);
+    expect(charizard);
   });
 
-  test("should protect the user from attacks", async () => {
+  it.each<{ numTurns: number; chance: number }>([
+    { numTurns: 1, chance: 3 },
+    { numTurns: 2, chance: 9 },
+    { numTurns: 3, chance: 27 },
+    { numTurns: 4, chance: 81 },
+  ])("should have a 1/$chance success rate after $times successful uses", async ({ numTurns, chance }) => {
```

Comment:
I goofed on the message lol...
```suggestion
  ])("should have a 1/$chance success rate after $numTurns successful uses", async ({ numTurns, chance }) => {
```